src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
```suggestion
  const children = people.filter(person => onlyWithSon
    ? person.sex === 'm' && people.some(mother => mother.name === person.mother)
    : people.some(mother => mother.name === person.mother)
  );
```

Reply:
Also, we do the same check(`person.sex === 'm'`...) in many places in the code
To make it easier to read and more clear, we can create functions for these checks.
For example we can create function `isMan` which will return `true` only if `person.sex === 'm'`
The same we can do with woman check and check if person has a mother.